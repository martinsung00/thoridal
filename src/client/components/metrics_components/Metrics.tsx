import * as React from "react";
import styles from "../../styles/metrics_compnent/Metrics.css";

interface Props {
  // To-do: Specify prop types
  net: number;
  totalWeekTrades: number;
  totalFulfilled: number;
  totalUnfulfilled: number;
  avgGainPerDay: number;
  avgTradePerDay: number;
}

export const Metrics: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <div className={styles.root}>
      <div className={styles.title}>Weekly Statistics</div>
      <div className={styles.left}>
        {props.net >= 0 && (
          <label className={styles.net}>
            NET GAIN: <div className={styles.gain}>+{props.net}</div>
          </label>
        )}
        {props.net < 0 && (
          <label className={styles.net}>
            NET LOSS: <div className={styles.loss}>{props.net}</div>
          </label>
        )}
        <div className={styles.divider} />
        <label className={styles.net}>
          TOTAL TRADES:{" "}
          <div className={styles.total}>{props.totalWeekTrades}</div>
        </label>
        <div className={styles.divider} />
        <div className={styles.divider} />
        <div className={styles.divider} />
        <label className={styles.secondary}>
          Total Completed Trades:{" "}
          <div className={styles.totalSecondary}>{props.totalFulfilled}</div>
        </label>
        <div className={styles.divider} />
        <label className={styles.secondary}>
          Total Pending Trades:{" "}
          <div className={styles.totalSecondary}>{props.totalUnfulfilled}</div>
        </label>
      </div>
      <div className={styles.right}>
        <label className={styles.perDay}>
          AVG GAIN PER DAY:{" "}
          <div className={styles.perDayTotal}>$ {props.avgGainPerDay}</div>
        </label>
        <div className={styles.divider} />
        <label className={styles.perDay}>
          AVG TRADE PER DAY:{" "}
          <div className={styles.perDayTotal}>{props.avgTradePerDay}</div>
        </label>
      </div>
    </div>
  );
};
