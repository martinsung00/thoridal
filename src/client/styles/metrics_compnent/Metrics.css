.root {
  position: relative;
  height: 23em;
  top: 9em;
  left: 3.7em;
  width: 90%;
  font-family: Helvetica, Arial, sans-serif;
  position: relative;
  border-top: 0.7px solid black;
  border-bottom: 0.7px solid black;
}

.title {
  padding-top: 1em;
  padding-bottom: 1em;
  text-align: center;
  font-size: 28px;
}

.left {
  position: relative;
  float: left;
  width: 25em;
  margin-left: 1em;
}

.right {
  position: relative;
  float: right;
  width: 25em;
  margin-right: 1.5em;
}

.net {
  position: relative;
  font-size: 26px;
  margin-top: 1em;
}

.gain {
  position: relative;
  float: right;
  color: rgb(0, 161, 0);
}

.loss {
  position: relative;
  float: right;
  color: red;
}

.total {
  position: relative;
  float: right;
  color: black;
}

.totalSecondary {
  position: relative;
  float: right;
  margin-right: 2em;
  color: rgb(104, 104, 104);
}

.secondary {
  position: relative;
  color: rgb(104, 104, 104);
}

.divider {
  height: 1.5em;
}

.perDay {
  position: relative;
  font-size: 26px;
  margin-top: 1em;
}

.perDayTotal {
  position: relative;
  float: right;
  color: black;
}
