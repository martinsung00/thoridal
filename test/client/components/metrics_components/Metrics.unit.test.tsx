import React from "react";
import renderer from "react-test-renderer";
import { Metrics } from "../../../../src/client/components/metrics_components/Metrics";

describe("Metrics Dashboard Tests", function () {
  const positiveNet = {
    net: 1,
    totalWeekTrades: 1,
    totalFulfilled: 1,
    totalUnfulfilled: 1,
    avgGainPerDay: 1,
    avgTradePerDay: 1,
  };

  const negativeNet = {
    net: -1,
    totalWeekTrades: 1,
    totalFulfilled: 1,
    totalUnfulfilled: 1,
    avgGainPerDay: -1,
    avgTradePerDay: 1,
  };

  it("should render all the elements with the given props (positive net)", function (done) {
    const container = renderer.create(<Metrics {...positiveNet} />).toJSON();
    expect(container).toMatchSnapshot();

    done();
  });

  it("should render all the elements with the given props (negative net)", function (done) {
    const container = renderer.create(<Metrics {...negativeNet} />).toJSON();
    expect(container).toMatchSnapshot();

    done();
  });
});
