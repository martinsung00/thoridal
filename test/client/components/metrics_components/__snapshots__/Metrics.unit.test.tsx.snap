// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics Dashboard Tests should render all the elements with the given props (negative net) 1`] = `
<div>
  <div>
    Weekly Statistics
  </div>
  <div>
    <label>
      NET LOSS: 
      <div>
        -1
      </div>
    </label>
    <div />
    <label>
      TOTAL TRADES:
       
      <div>
        1
      </div>
    </label>
    <div />
    <div />
    <div />
    <label>
      Total Completed Trades:
       
      <div>
        1
      </div>
    </label>
    <div />
    <label>
      Total Pending Trades:
       
      <div>
        1
      </div>
    </label>
  </div>
  <div>
    <label>
      AVG GAIN PER DAY:
       
      <div>
        $ 
        -1
      </div>
    </label>
    <div />
    <label>
      AVG TRADE PER DAY:
       
      <div>
        1
      </div>
    </label>
  </div>
</div>
`;

exports[`Metrics Dashboard Tests should render all the elements with the given props (positive net) 1`] = `
<div>
  <div>
    Weekly Statistics
  </div>
  <div>
    <label>
      NET GAIN: 
      <div>
        +
        1
      </div>
    </label>
    <div />
    <label>
      TOTAL TRADES:
       
      <div>
        1
      </div>
    </label>
    <div />
    <div />
    <div />
    <label>
      Total Completed Trades:
       
      <div>
        1
      </div>
    </label>
    <div />
    <label>
      Total Pending Trades:
       
      <div>
        1
      </div>
    </label>
  </div>
  <div>
    <label>
      AVG GAIN PER DAY:
       
      <div>
        $ 
        1
      </div>
    </label>
    <div />
    <label>
      AVG TRADE PER DAY:
       
      <div>
        1
      </div>
    </label>
  </div>
</div>
`;
